admin-dev/themes/default/template/controllers/modules/modal_translation.tpl

```diff
@@ -31,7 +31,22 @@
 		</button>
 		<ul class="dropdown-menu">
 			{foreach from=$module_languages item=language}
-			<li><a href="{$trad_link|escape:'html':'UTF-8'}{$language['iso_code']|escape:'html':'UTF-8'}#{$module_name|escape:'html':'UTF-8'}">{$language.name|escape:'html':'UTF-8'}</a></li>
+				<li>
+					{if $new_translate_system }
```

Comment:
I'm not a big fan of all this logical in smarty. I would rather have an array of links built in the PHP.
Something like 
```php
$translateLinks = array(
    'fr' => $link->getAdminLink...,
    'en' => $link->getAdminLink...,
);
```
and in your smarty
```
$translateLinks[$language.iso_code]
```